core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3144,15 +3144,34 @@ public void activeSyncMetadata(AlluxioURI path, Collection<AlluxioURI> changedFi
 
   @Override
   public boolean recordActiveSyncTxid(long txId, long mountId) {
-    try (RpcContext rpcContext = createRpcContext()) {
+    MountInfo mountInfo = mMountTable.getMountInfo(mountId);
+    AlluxioURI mountPath = null;
+    if (mountInfo != null) {
+      mountPath = mountInfo.getAlluxioUri();
+    } else {
+      return false;
+    }
 
+    try (RpcContext rpcContext = createRpcContext();
+         LockedInodePath inodePath = mInodeTree
+             .lockFullInodePath(mountPath, LockPattern.READ)) {
```

Comment:
if the point of locking the path is to ensure that nothing removes the mount before journaling the txid, we need to check while holding the lock that the mount point still exists.

I see there's a `mSyncManagerLock` in `ActiveSyncManager` - do you think that would be a more appropriate guard for changing sync points? If we use the inode tree, we need to make sure that anything which reads/modifies sync points is guarded by an inode lock